### DISCUSSION TITLE
Is there a way to register Alpine.data / magic / directive / bind, etc after Alpine.start()?

### DISCUSSION THREAD

**Discussion by jonnitto** [alpinejs/alpine#4024] [1. Help]
If you work with an environment where you swapping content (aka. HTML over the wire), it would be much easier to have the possibility to add  Alpine.data / magic / directive / bind, etc after Alpine.start().

Is there any way to solve this?

**Comment by danie-ramdhani** [comment]
is `alpine:init` what you're looking for?

```html
...
    <script>
        document.addEventListener('alpine:init', () => {
            Alpine.data('fooData', () => ({
                fooText: 'foo',
            }));
        });
    </script>

</head>
```

then:

```html
    <div x-data="fooData">
        <span x-text="fooText"></span>
    </div>
```

**Comment by jonnitto** [comment]
Not really. `alpine:init` runs when alpine get initialized (just as the name says). I would like to have solution where I can register it after alpine js has started.

**Comment by ekwoka** [comment]
You can register them after it starts as is.

You just can't have put the elements that depend on them onto the page yet.

**Reply by jonnitto** [reply]
Hm. I'll try this
